pkg/ds/queue/queue.go

```diff
@@ -0,0 +1,119 @@
+package queue
+
+type Node[T any] struct {
+	value  T
+	next   *Node[T]
+	before *Node[T]
+}
+
+type Queue[T any] struct {
+	head *Node[T]
+	tail *Node[T]
+}
```

Comment:
Add `size` to the `Queue` and use it for checking empty list and for allocating slice in method `List`.

```suggestion
type Queue[T any] struct {
	head *Node[T]
	tail *Node[T]
	size int
}
```

Reply:
updated in https://github.com/KyberNetwork/tradinglib/pull/13/commits/c70340a02facf62735fa1cd710d2f2bd8d8338d1